lib/Dialect/TritonGPU/Transforms/Pipeliner/SoftwarePipeliner.cpp

```diff
@@ -115,8 +121,11 @@ struct PipelinePass : public impl::TritonGPUPipelineBase<PipelinePass> {
         loops.push_back(forOp);
     });
 
-    if (loops.empty())
+    if (loops.empty()) {
+      auto op = getOperation();
+      op->emitRemark() << "Warning: SWP fails. There is no loop with num_stages greater than 1";
```

Comment:
If there is no loop with num_stages greater than 1, I guess we don't need to email a warning since SWP is not requested. For generating test cases, maybe we can go through TritonBench or Triton lit tests to see if a remark will be triggered.